src/test/java/uk/gov/hmcts/reform/ucmc/service/PaymentsServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.ucmc.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.payments.client.PaymentsClient;
+import uk.gov.hmcts.reform.payments.client.models.FeeDto;
+import uk.gov.hmcts.reform.payments.client.models.PaymentDto;
+import uk.gov.hmcts.reform.payments.client.request.CreditAccountPaymentRequest;
+import uk.gov.hmcts.reform.ucmc.config.PaymentsConfiguration;
+import uk.gov.hmcts.reform.ucmc.request.RequestData;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(SpringExtension.class)
+class PaymentsServiceTest {
+    private static final String SERVICE = "service";
+    private static final String SITE_ID = "site_id";
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final FeeDto FEE_DATA = FeeDto.builder()
+        .version("1")
+        .code("CODE")
+        .calculatedAmount(BigDecimal.ONE)
+        .build();
+    private static final PaymentDto PAYMENT_DTO = PaymentDto.builder().reference("RC-1234-1234-1234-1234").build();
+
+
+    @Mock
+    private FeesService feesService;
+
+    @Mock
+    private PaymentsClient paymentsClient;
+
+    @Mock
+    private RequestData requestData;
+
+    @Mock
+    private PaymentsConfiguration paymentsConfiguration;
+
+    @InjectMocks
+    private PaymentsService paymentsService;
+
+    @BeforeEach
+    void setUp() {
+        given(feesService.getFeeDataByClaimValue(any())).willReturn(FEE_DATA);
+        given(paymentsClient.createCreditAccountPayment(any(), any())).willReturn(PAYMENT_DTO);
+        given(requestData.authorisation()).willReturn(AUTH_TOKEN);
+        given(paymentsConfiguration.getService()).willReturn(SERVICE);
+        given(paymentsConfiguration.getSiteId()).willReturn(SITE_ID);
+    }
+
+    @Test
+    public void shouldCreateCreditAccountPayment() {
+        CaseDetails caseDetails = CaseDetails.builder()
```

Comment:
you see any value in using json fixture files for things like this?

Reply:
Maybe if the data structure is really large. It's good to have everything there in the code, but I don't really have strong opinion on that.

Reply:
Builders will have advantage of catching some issues at compile time i.e. renaming of fields etc. May be a utility class to generate the similar kind of data and we can reuse where we can. Just a suggestion 🤔